packages/contracts-bedrock/contracts/L1/OptimismPortal.sol

```diff
@@ -162,32 +162,39 @@ contract OptimismPortal is Initializable, ResourceMetering, Semver {
     }
 
     /**
-     * @notice Determine if an L2 Output is finalized.
+     * @notice Determine if an L2 Output is finalized for a given block number.
      *
      * @param _l2BlockNumber The number of the L2 block.
      */
+    function isOutputFinalized(uint256 _l2BlockNumber) public view returns (bool) {
+        L2OutputOracle.OutputProposal memory proposal;
 
-    function isOutputFinalized(uint256 _l2BlockNumber) external view returns (bool) {
-        L2OutputOracle.OutputProposal memory proposal = L2_ORACLE.getL2Output(_l2BlockNumber);
+        uint256 interval = L2_ORACLE.SUBMISSION_INTERVAL();
+        uint256 startingBlockNumber = L2_ORACLE.STARTING_BLOCK_NUMBER();
+        if (startingBlockNumber > _l2BlockNumber) {
+            return false;
+        }
+
+        // Find the distance between _l2BlockNumber, and the checkpoint block before it.
+        uint256 offset = (_l2BlockNumber - startingBlockNumber) % interval;
```

Comment:
I think this math makes more sense in `getL2Output`, since I would also want to use it on the client-side and this saves me doing some math.